pallets/parachain-staking/src/delegation_requests.rs

```diff
@@ -0,0 +1,317 @@
+{
+	/// Returns the wrapped amount value.
+	pub fn amount(&self) -> Balance {
+		match self {
+			DelegationAction::Revoke(amount) => *amount,
+			DelegationAction::Decrease(amount) => *amount,
+		}
+	}
+}
+
+/// Represents a scheduled request that define a [DelegationAction]. The request is executable
+/// iff the provided [RoundIndex] is achieved.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct ScheduledRequest<Balance> {
+	pub when_executable: RoundIndex,
+	pub action: DelegationAction<Balance>,
+}
+
+impl<T> Pallet<T>
+where
+	T: Config,
+{
+	/// Schedules a [DelegationAction::Revoke] for the delegator, towards a given collator.
+	pub(crate) fn delegator_schedule_revoke(
+		delegator: DelegatorId<T>,
+		collator: CollatorId<T>,
+	) -> DispatchResultWithPostInfo {
+		let state = <DelegatorState<T>>::get(&delegator).ok_or(<Error<T>>::DelegatorDNE)?;
+		ensure!(
+			<DelegatorScheduledRequests<T>>::get(&delegator, &collator).is_none(),
+			<Error<T>>::PendingDelegationRequestAlreadyExists,
+		);
+
+		let bonded_amount = state
+			.get_bond_amount(&collator)
+			.ok_or(<Error<T>>::DelegationDNE)?;
+		let now = <Round<T>>::get().current;
+		let when = now + T::RevokeDelegationDelay::get();
+
+		Self::delegator_scheduled_requests_state_add(
+			delegator.clone(),
+			collator.clone(),
+			ScheduledRequest {
+				action: DelegationAction::Revoke(bonded_amount),
+				when_executable: when,
+			},
+		);
+
+		Self::deposit_event(Event::DelegationRevocationScheduled {
+			round: now,
+			delegator,
+			candidate: collator,
+			scheduled_exit: when,
+		});
+		Ok(().into())
+	}
+
+	/// Schedules a [DelegationAction::Decrease] for the delegator, towards a given collator.
+	pub(crate) fn delegator_schedule_bond_decrease(
+		delegator: DelegatorId<T>,
+		collator: CollatorId<T>,
+		decrease_amount: BalanceOf<T>,
+	) -> DispatchResultWithPostInfo {
+		let state = <DelegatorState<T>>::get(&delegator).ok_or(<Error<T>>::DelegatorDNE)?;
+		ensure!(
+			<DelegatorScheduledRequests<T>>::get(&delegator, &collator).is_none(),
+			<Error<T>>::PendingDelegationRequestAlreadyExists,
+		);
+
+		let bonded_amount = state
+			.get_bond_amount(&collator)
+			.ok_or(<Error<T>>::DelegationDNE)?;
+
+		ensure!(
+			bonded_amount > decrease_amount,
+			<Error<T>>::DelegatorBondBelowMin
+		);
+		let new_amount: BalanceOf<T> = (bonded_amount - decrease_amount).into();
+		ensure!(
+			new_amount >= T::MinDelegation::get(),
+			<Error<T>>::DelegationBelowMin
+		);
+
+		// Net Total is total after pending orders are executed
+		let total_scheduled_decrease_amount =
+			<DelegatorScheduledRequestDecreaseAmount<T>>::get(&delegator);
+		let net_total = state.total.saturating_sub(total_scheduled_decrease_amount);
+
+		// Net Total is always >= MinDelegatorStk
+		let max_subtracted_amount = net_total.saturating_sub(T::MinDelegatorStk::get().into());
+		ensure!(
+			decrease_amount <= max_subtracted_amount,
+			<Error<T>>::DelegatorBondBelowMin
+		);
```

Comment:
This is where the `DelegatorScheduledRequestDecreaseAmount` is used. It is accessed after `DelegatorState` so it is better to move it into `DelegatorState`.

I need to look to see if we increase it in the body of `delegator_schedule_requests_state_add` and write to storage.

Reply:
For context: 

If we move the field back, we get:

* schedule / cancel / execute:
  - 1 read of DelegatorState
  - 1 write of DelegatorState

And if we keep it in separate storage:

* schedule / cancel:
  - 1 read of DelegatorState
  - 1 read of DelegatorScheduledRequestDecreaseAmount
  - 1 write of DelegatorScheduledRequestDecreaseAmount

* execute:
  - 1 read of DelegatorState
  - 1 write of DelegatorState (mutation of another field)
  - 1 write of DelegatorScheduledRequestDecreaseAmount

The choice is always between `higher` but `smaller` read/writes vs. `lower` but `bigger` read/writes.

This scenario is quite straightforward and not of much importance, but we'd like to do the write thing for future PoV/performance.

/cc @crystalin @notlesh @librelois